src/main/kotlin/com/cognifide/gradle/aem/common/instance/InstanceFactory.kt

```diff
@@ -15,10 +15,12 @@ class InstanceFactory(val aem: AemExtension) {
     fun remote(httpUrl: String, configurer: Instance.() -> Unit = {}): Instance {
         return Instance(aem).apply {
             val instanceUrl = InstanceUrl.parse(httpUrl)
+            val bearerToken = InstanceIMSClient.generateBearerTokenOrNull(aem)
```

Comment:
hmm... this causes that 

p - number of gradle proejcts / modules
i - number of instances

(p * i) times token will be retrieved. I am not sure if this will not have serious/dangerous implications

Reply:
I am wondering if we maybe rather should generate that token once per instance; I have a BuildScope class for such cases

Reply:
Extracted that one out of the remote method so to be generated once